Add data subscription document (#400)
apache/iotdb-docs: src/UserGuide/Master/Deployment-and-Maintenance/Data-subscription.md

2. **Rapid Integration with Downstream Systems**: Easier to integrate with system components such as Flink, Spark, Kafka / DataX, Camel / MySQL, PG, etc. There is no need to develop custom logic for IoTDB change capture for each big data component, which simplifies the design of integration components and improves development efficiency.

## 2. Key Concepts

The IoTDB Subscription Client encompasses three core concepts: Topic, Consumer, and Consumer Group. The specific relationships are illustrated in the diagram below:

<div align="center">
    <img src="https://alioss.timecho.com/docs/img/Data_sub_01.png" alt="" style="width: 60%;"/>
</div>

1. **Topic**: A Topic is a logical concept used in the IoTDB Subscription Client to categorize data, which can be considered as a data publication channel. Producers publish data to specific topics, and consumers subscribe to these topics to receive relevant data. Unlike Kafka, in the IoTDB subscription client, topics correspond to the sequence and time range to be subscribed to, output format (message or TsFile), and optional custom processing logic.

> 2. **Consumer**: Consumer is the application or service where the IoTDB subscription client is located, responsible for receiving and processing data published to specific topics. Consumers retrieve data from the queue and process it accordingly. There are two types of Consumers available in the IoTDB subscription client:

Reviewer: ```suggestion
2. **Consumer**: Consumer is an IoTDB subscription client, responsible for receiving and processing data published to specific topics. Consumers retrieve data from the queue and process it accordingly. There are two types of Consumers available in the IoTDB subscription client:
```
